deep_qa/data/datasets/dataset.py

```diff
@@ -71,23 +85,16 @@ def to_indexed_dataset(self, data_indexer: DataIndexer) -> 'IndexedDataset':
         return IndexedDataset(indexed_instances)
 
     @staticmethod
-    def read_from_file(filename: str, instance_class):
+    def read_from_file(filename: str, instance_class, params: Params=None):
         with codecs.open(filename, 'r', 'utf-8') as input_file:
             lines = [x.strip() for x in tqdm.tqdm(input_file.readlines())]
-            return TextDataset.read_from_lines(lines, instance_class)
+            return TextDataset.read_from_lines(lines, instance_class, params)
 
     @staticmethod
-    def read_from_lines(lines: List[str], instance_class):
+    def read_from_lines(lines: List[str], instance_class, params: Params=None):
         instances = [instance_class.read_from_line(x) for x in lines]
-        labels = [(x.label, x) for x in instances]
-        labels.sort(key=lambda x: str(x[0]))
-        label_counts = [(label, len([x for x in group]))
-                        for label, group in itertools.groupby(labels, lambda x: x[0])]
-        label_count_str = str(label_counts)
-        if len(label_count_str) > 100:
-            label_count_str = label_count_str[:100] + '...'
-        logger.info("Finished reading dataset; label counts: %s", label_count_str)
-        return TextDataset(instances)
+        count_labels(instances)
```

Comment:
The name here looks like it should return something, so seeing it with an ignored return value seems odd.  Maybe call this `print_label_counts`, or `log_label_counts`?